### PR TITLE
Remove fk_product_fournisseur_price_fk_product

### PR DESCRIPTION
**Pull request by wdammak**
Delete a composite key that makes no sense.

### CODE REVIEW

File: htdocs/install/mysql/migration/9.0.0-10.0.0.sql [hunk 1]
```diff
@@ -254,3 +254,4 @@ ALTER TABLE llx_expensereport_det ADD COLUMN subprice double(24,8) DEFAULT 0 NOT
 ALTER TABLE llx_product_attribute_combination ADD INDEX idx_product_att_com_product_parent (fk_product_parent);
 ALTER TABLE llx_product_attribute_combination ADD INDEX idx_product_att_com_product_child (fk_product_child);
 
+ALTER TABLE llx_product_fournisseur_price DROP FOREIGN KEY fk_product_fournisseur_price_fk_product;
```
Comment (eldy) [inline htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: Why removing this integrity key ? This key is defined has integrity on product with

ALTER TABLE llx_product_fournisseur_price ADD CONSTRAINT fk_product_fournisseur_price_fk_product FOREIGN KEY (fk_product) REFERENCES llx_product (rowid);

Removing it means we agree to have prices in this table for product that does not exists anymore.

Comment (wdammak) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: What I don't understand, why we can not have several prices (identical or different) from the same supplier and for the same product?

Why it's not the same principle for purchase prices (product_price)?

Comment (eldy) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: You can have several prices for same supplier and product.
For exemple the product P is sold by supplier S with price x and product P is sold by supplier S with price y.
Then into table we will have 
fk_product=P, fk_soc=S and ref_fourn = 'P-S-1'   for example (first reference)
and
fk_product=P, fk_soc=S and ref_fourn = 'P-S-2'   for example (second reference with another price)
The unique key already allow this.

Don't you make a confusion between unique key and foreign key ?

Comment (wdammak) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: @eldy
It's common to have:
price1/date1 for the same product, same company and same ref_fourn => fk_product = P, fk_soc = S and ref_fourn = 'P-S-1'
price1/date2 for the same product, same company and same ref_fourn => fk_product = P, fk_soc = S and ref_fourn = 'P-S-1'
price2/date3 for the same product, same company and same ref_fourn => fk_product = P, fk_soc = S and ref_fourn = 'P-S-1'

I just made tests directly in the database, the insert works correctly ?!
Maybe the problem comes from the web interface or something that escapes me ...

Comment (wdammak) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: @eldy
I have just done the test via the web interface (Add new price of purchase) it updates the existing row with the new price in product_provider_price and also inserts it in product_provider_price_log

what is the purpose of these two tables and why it's not manage in the same table product_provider_price?

Comment (wdammak) [reply htdocs/install/mysql/migration/9.0.0-10.0.0.sql:257]: @eldy 
and how to consult the olds purchase prices in dolibarr?

**Comment by eldy** [comment]
There is a table llx_product_fournisseur_price_log dedicated to store old prices.

**Pull request closed without merging**